src/shared/store/filters/filter.selectors.ts

```diff
@@ -39,7 +39,7 @@ export namespace FilterSelectors {
 
     switch (sortingMethod) {
       case SortingOption.default:
-        sortedArray.sort((a: ICalculatedProduct, b: ICalculatedProduct) => a.id - b.id);
+        sortedArray.sort((a: ICalculatedProduct, b: ICalculatedProduct) => a.id.localeCompare(b.id));
```

Comment:
What if we use a more general callback function here? 
` sortedArray.sort((a: ICalculatedProduct, b: ICalculatedProduct) => a.id <= b.id  ?  -1 :  1 );`
It would work, whatever happens with the type of id, if we change the db again in the future.

Reply:
When we use the less-than (<) and greater-than (>) operators to compare strings, JavaScript compares them lexicographically. For these IDs, the lexicographic order may not be the same as the desired order, especially if the IDs don't follow a simple pattern.

To sort such IDs correctly, we should use a more suitable comparison function. The localeCompare method is designed for comparing strings and considers the correct order based on the characters' Unicode values, which makes it suitable for sorting strings, including complex ones like IDs.

I also found this on the developer.mozilla site:
Warning: Do not rely on exact return values of -1 or 1!

Negative and positive integer results vary between browsers (as well as between browser versions) because the ECMAScript specification only mandates negative and positive values. Some browsers may return -2 or 2, or even some other negative or positive value.